R/Mexico.R

```diff
@@ -141,20 +147,15 @@ Mexico <- R6::R6Class("Mexico",
     #'
     clean_level_2 = function() {
       self$data$clean <- self$data$raw %>%
-        mutate(
-          region_level_2 = .data$nombre,
-          inegi_state = substr(.data$cve_ent, 1, 2),
-          date = dmy(.data$date)
-        ) %>%
-        select(-.data$nombre) %>%
-        full_join(self$data$codes_lookup, by = "inegi_state") %>%
-        mutate(
-          level_1_region_code = .data$iso_code,
-          level_2_region_code = .data$cve_ent
-        ) %>%
+        rename(level_2_region = .data$nombre) %>%
+        mutate(date = dmy(.data$date)) %>%
+        left_join(self$region_codes %>%
+                    filter(!is.na(level_2_region)),
+                  by = "level_2_region") %>%
```

Comment:
great stuff. Can we just use a unique filter to stop duplication?

Reply:
(if we think its in the raw data)

Reply:
i can't figure out where it comes from, but yep will add a distinct()

Reply:
if you can't find it its probably an issue in the raw data?